unstructured/ingest/doc_processor/generalized.py

```diff
@@ -30,9 +32,11 @@ def process_document(doc):
         doc.write_result()
 
     except Exception:
+        filename = doc.filename if isinstance(doc.filename, Path) else Path(doc.filename)
+        if filename.exists():
+            os.unlink(filename.as_posix())
         # TODO(crag) save the exception instead of print?
-        logging.error(f"Failed to process {doc}", exc_info=True)
-    else:
-        doc.cleanup_file()
+        logger.error(f"Failed to process {doc}", exc_info=True)
     finally:
+        doc.cleanup_file()
```

Comment:
Note that removing the file-deletion in the `except Exception` above is the same thing as keeping this line, since the code right after the `finally` will always be executed, so please LMK what option you prefer:
* The implemented one
* Or, the one that just moves the `doc.cleanup_file` from `else` to `finally` and doesn't perform the `filename.exists()` and `os.unlink()` under the `except`